docs/Reference/Server.md

```diff
@@ -124,6 +125,19 @@ use. Also, when `serverFactory` option is specified, this option is ignored.
 
 + Default: `5000` (5 seconds)
 
+### `forceCloseConnections`
+<a id="forcecloseconnections"></a>
+
+When set to `true` requests with the header `connection: keep-alive` will be
+tracked by the server. Upon [`close`](#close), the server will iterate the
+current persistent connections and [destroy their
+sockets](https://nodejs.org/dist/latest-v16.x/docs/api/net.html#socketdestroyerror).
+This means the server will shutdown immediately instead of waiting for existing
+persistent connections to timeout first. Important: connections are not
+inspected to determine if requests have been completed.
```

Comment:
Do you think it could be possible to order the shutdown cycle in this way:

 1. Disable acceptance of new requests
 2. In-progress request handlers are allowed to complete
 3. Once all handlers have completed, close the sockets with keep-alive connections?

Reply:
As far as I know, it is impossible to determine if a request is pending via the socket. We only retain the sockets so that we do not keep the full request(s) in memory for the duration of the persistent connection.

Reply:
Node.js knows btw, however that info is private

Reply:
> As far as I know, it is impossible to determine if a request is pending via the socket.

Wouldn't we know via fastify tracking `handler()`s that are pending, rather than the socket?

> We only retain the sockets so that we do not keep the full request(s) in memory for the duration of the persistent connection.

Yeah, I'm talking about while the request is still active and the connection is active.